app/views/shops/_form.html.haml

```diff
@@ -25,4 +25,23 @@
 			= f.label :email_id
 			= f.email_field :email_id
 			%br/
-			= f.submit(class: "btn btn-primary")
\ No newline at end of file
+			Location:
+			%input#address{:style => "width: 200px", :type => "text"}/
+			#map{:style => "width: 500px; height: 400px;"}
+			= f.hidden_field :latitude
+			= f.hidden_field :longitude
+			:javascript
```

Comment:
You can overwrite the field's ID. 

`= f.hidden_field :latitude, id: 'latitude'`

Or if you want to make it more dynamic, have a `#map_wrapper` and in your JS code you find the lat and lng hidden fields within that. But do the first option is much easier and clearer.